docs/assumptions.rst

```diff
@@ -66,21 +183,50 @@ file. A typical checkpoint will look like:
   close(fd)
   chmod("checkpoint1.chk", 0444)
 
-Future versions of UnifyFS may support different laminate semantics, such as
-laminate on close() or laminate via an explicit API call.
+We plan for future versions of UnifyFS to support different methods for
+laminating files, such as
+laminating all files on close() or laminating via an explicit API call.
 
 We define the laminated consistency model to enable certain optimizations while
-supporting the perceived requirements of application checkpoints.  Since remote
-processes are not permitted to read arbitrary bytes of a file until its
-lamination, UnifyFS can buffer all data and metadata of the file locally
-(instead of exchanging indexing information between compute nodes) before the
-lamination occurs.  Also, since file contents cannot change after lamination,
+supporting the typical requirements of bulk synchronous I/O.
+Recall that for bulk synchronous I/O patterns, reads and writes typically occur in
+distinct phases. This means that for the majority of the time,
+processes do not need to read arbitrary
+bytes of a file until the write phase is completed, which in practice is
+when the file is done being modified and closed and can be safely made
+read-only with lamination.
+By assuming that processes do not need to access file data modified
+by other processes before lamination,
```

Comment:
Maybe tweak this to highlight that "well-behaved" apps can benefit from the extreme write buffering optimization.
```suggestion
For applications in which processes do not need to access file data modified
by other processes before lamination,
```

Reply:
+1